power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/utils.hpp

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_UTILS_HPP
+#define POWER_GRID_MODEL_CPP_UTILS_HPP
+
+#include "basics.hpp"
+#include "handle.hpp"
+
+#include <algorithm>
+#include <complex>
+#include <limits>
+#include <type_traits>
+
+namespace power_grid_model_cpp {
+template <typename T>
+inline typename std::enable_if<std::is_floating_point<T>::value, // NOLINT(modernize-use-constraints,
+                                                                 // modernize-type-traits)
+                               bool>::type
```

Comment:
now that i think about it, you may be able to disable the type hint warning by setting the `CXX_STANDARD` target property in CMake (https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html and https://cmake.org/cmake/help/latest/command/set_target_properties.html)

```cmake
set_target_properties(power_grid_model_cpp PROPERTIES CXX_STANDARD 17)
```

not sure about that, though. If it does not solve the problem, please keep as is.

Reply:
You cannot do that. In your test code which you consume the wrapper header, you specify c++20.

So this will be conflict.